src/main/java/com/Mattheo992/schoolapplication/Controller/StudentController.java

```diff
@@ -0,0 +1,44 @@
+package com.Mattheo992.schoolapplication.Controller;
+
+import com.Mattheo992.schoolapplication.Model.Student;
+import com.Mattheo992.schoolapplication.Service.StudentService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/students")
+public class StudentController {
+private final StudentService studentService;
+
+@GetMapping
+    public List<Student> getStudents() {
+        return studentService.getStudents();
+    }
+@GetMapping("/id/{id}")
+    public Student getStudentById(@PathVariable Long id) {
+        return studentService.getStudentById(id);
+    }
+@GetMapping("/{lastName}")
+    public List<Student> getStudentsWithTheSameLastName(@PathVariable String lastName) {
+        return studentService.getStudentsWithTheSameLastName(lastName);
+    }
+@PostMapping
+    public void createStudent(@RequestBody Student student) {
```

Comment:
Ja bym zwrócił Studenta -> powołam się tutaj na przykład z ID. Aplikacja nadaje id w sytuacji jesli aplikacja A komunikowalaby sie z aplikacja B i A rzucilo request o utworzenie studenta to w odpowiedzi nie dostalaby id studenta, który został utworzony wiec nie bylo by w stanie dojsc jakiego studenta utworzylo